pkg/kubelet/network/generic/generic.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generic
+
+import (
+	"errors"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/golang/glog"
+)
+
+type genericNetworkPlugin struct {
+	host network.Host
+}
+
+const (
+	genericPluginName = "generic"
+	initCmd           = "--init"
+	preLaunchCmd      = "--pre-pod-launch"
+	preDeleteCmd      = "--pre-pod-delete"
+	executable        = "/usr/local/bin/kubelet-plugins/net/generic/generic"
```

Comment:
latest standard or not, Ubuntu 14.04 doesn't use it.  That doesn't mean we
can't, it just means that we'll have to mkdir() the whole thing :)

On Tue, Mar 17, 2015 at 9:44 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/kubelet/network/generic/generic.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5069#discussion_r26637964>
> :
>
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
> > +	"github.com/golang/glog"
> > +)
> > +
> > +type genericNetworkPlugin struct {
> > +	host network.Host
> > +}
> > +
> > +const (
> > +	genericPluginName = "generic"
> > +	initCmd           = "--init"
> > +	preLaunchCmd      = "--pre-pod-launch"
> > +	preDeleteCmd      = "--pre-pod-delete"
> > +	executable        = "/usr/local/bin/kubelet-plugins/net/generic/generic"
>
> /usr/libexec/<app> appears to be the latest standard:
> http://www.linuxbase.org/betaspecs/fhs/fhs/ch04s07.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5069/files#r26637964>
> .
>